azure/azjwt/azjwt.go

```diff
@@ -147,12 +148,24 @@ func (a *Auth) Authenticator(next http.Handler) http.Handler {
 			return
 		}
 
+		// Add the email to the context so that it can be used by the handler
+		ctx := context.WithValue(r.Context(), emailContextKey{}, allowedEmails[0])
 		// Token is authenticated, pass it through
-		next.ServeHTTP(w, r)
+		next.ServeHTTP(w, r.WithContext(ctx))
 	}
 	return http.HandlerFunc(hfn)
 }
 
+type emailContextKey struct{}
+
+func EmailFromContext(ctx context.Context) (string, error) {
```

Comment:
I'd just use `func (context.Context) (string, bool)` here, the only realistic failure is "there was no email"